packages/psss/src/containers/Panels/WeeklyReportsPanel.js

```diff
@@ -124,38 +130,54 @@ const toCommaList = values =>
     .toUpperCase()
     .replace(/,(?!.*,)/gim, ' and');
 
+const getSyndromeAlerts = (countryData, weekNumber) => {
+  if (countryData.length === 0 || !weekNumber) {
+    return [];
+  }
+
+  return countryData[weekNumber].syndromes.reduce(
+    (statuses, syndrome) => (syndrome.isAlert ? [...statuses, syndrome] : statuses),
+    [],
+  );
+};
+
 export const WeeklyReportsPanelComponent = React.memo(
-  ({ countryData, sitesData, isOpen, handleClose, unVerifiedSyndromes, alerts, handleConfirm }) => {
+  ({ countryData, isOpen, handleClose, weekNumber, unVerifiedSyndromes }) => {
     const [panelStatus, setPanelStatus] = useState(PANEL_STATUSES.INITIAL);
     const [countryTableStatus, setCountryTableStatus] = useState(TABLE_STATUSES.STATIC);
     const [sitesTableStatus, setSitesTableStatus] = useState(TABLE_STATUSES.STATIC);
     const [activeSiteIndex, setActiveSiteIndex] = useState(0);
     const [isModalOpen, setIsModalOpen] = useState(false);
     const { countryCode } = useParams();
+    const { data: sitesData } = useTableQuery('sites', {
+      countryCode,
+      weekNumber,
+    });
 
-    const isVerified = unVerifiedSyndromes.length === 0;
-    const hasAlerts = alerts.length > 0;
-
-    const handleSubmit = useCallback(() => {
-      if (isVerified) {
-        handleConfirm();
-
-        if (hasAlerts) {
+    const [confirmReport] = useMutation(() => FakeAPI.post(), {
+      onSuccess: () => {
+        if (getSyndromeAlerts(countryData, weekNumber)) {
           setIsModalOpen(true);
         }
-
         setPanelStatus(PANEL_STATUSES.SUCCESS);
+        queryCache.invalidateQueries('country-weeks', { countryCode, weekNumber });
```

Comment:
I'm wondering if there's a better way to manage what queries get invalidate by what mutations. E.g. a mutation to a single site's data would mutate the site query as well as the country query. Seems like the kind of thing we'd want to have centrally somehow, because down at the component level isn't really the place to be thinking of "what other parts of the app will be affected by this change"